gsrs-spring-boot-autoconfigure/src/main/java/gsrs/controller/AbstractLegacyTextSearchGsrsEntityController.java

```diff
@@ -189,7 +214,171 @@ public ResponseEntity<Object> searchV1(@RequestParam("q") Optional<String> query
         ResponseEntity<Object> ret= new ResponseEntity<>(createSearchResponse(results, result, request), HttpStatus.OK);
         return ret;
     }
+    
+    @PostGsrsRestApiMapping(value="/@bulkQuery")
+    public ResponseEntity<String> saveQueryList(@RequestBody String query,
+    									@RequestParam("top") Optional<Integer> top,
+  										@RequestParam("skip") Optional<Integer> skip,
+  										HttpServletRequest request){
+    	
+    	int qTop = 100, qSkip = 0;
+    	if(top.isPresent()) 
+    		qTop = top.get();
+    	if(skip.isPresent()) 
+    		qSkip = top.get();
+    	    	
+    	List<String> queries = Arrays.asList(query.split("\\r?\\n|\\r"));
+  	  
+    	List<String> list = queries.stream()    			
+    			.map(q->q.trim())
+    			.filter(q->q.length()>0)
+    			.distinct()
```

Comment:
Just an FYI, Larry might prefer that the list not be reduced this way with distinct due to work flow. When one is doing matching work we often have lists of substances say in an exel spreadsheet. There could be duplicates in that list but if the row is omitted from the result, the list in excel will no longer be the same size as what came back from GSRS.

On the other hand, I understand you don't want to do the same searches over and over.

Reply:
Thanks, Alex. That is a good point. That makes me think about the the order in the summary result. Is is expected that we preserve the order from the original query list? No matter they are from a file or not.